fate/src/main/java/org/apache/accumulo/fate/zookeeper/ZooCache.java

```diff
@@ -225,6 +223,9 @@ public T retry() {
           if (childrenCache.containsKey(zPath)) {
             return childrenCache.get(zPath);
           }
+
+          final ZooKeeper zooKeeper = getZooKeeper();
```

Comment:
Pushing this down into the ZooRunnable does have the potential for future programming errors. Implementations could try to "be efficient" and call `getZooKeeper()` once and store it as a member variable to the `ZooRunnable` implementation. This would negate the reconstruction of the `ZooKeeper` instance that the `retry()` loop is doing.

I think some javadoc explaining that `getZooKeeper()` could be used by the implementations on the `abstract T run()` definition for future Accumulo devs would be very worthwhile.